pkg/store/store.go

```diff
@@ -0,0 +1,85 @@
+package store
+
+import (
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// WatchCallback is used for watch methods on keys
+// and is triggered on key change
+type WatchCallback func(value [][]byte)
```

Comment:
I think there is only one index for the whole tree/key so in the case of the WatchCallback returning this index could be sufficient?